### PR TITLE
[ENH] OWBoxPlot: Show missing values

### LINKED ISSUE #4128: Box Plot: show missing values
**Is your feature request related to a problem? Please describe.**
Box Plot doesn't show missing values for discrete data.

**Describe the solution you'd like**
Sometimes, the user would like to see the proportion of missing values for a certain attribute and box plot is a nice visualization for this. Color the missing values grey.

**Describe alternatives you've considered**
Feature Statistics shows this exact information. Perhaps it is enough.

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Issue

Fixes #4128

##### Description of changes
- In this PR we extend the box plot widget to show missing values for discrete variables.
- We update the contingency to have three additional parameters to hold number of unknowns for each column, row and combined unknowns. Also, calls of contingency around Orange were updated.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/statistics/distribution.py [hunk 1]
```diff
@@ -211,6 +211,11 @@ def max(self):
         if self.variable.ordered:
             return self.variable.values[-1]
 
+    # def sum(self, *args, **kwargs):
+    #     res = self.sum(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (PrimozGodec) [reply Orange/statistics/distribution.py]: Ok, that was my stupid mistake. Thank you. Sure it works.

**Comment by janezd** [comment]
We discussed this. Yes it would make sense to group by having/not having missing values. Good idea.

**Comment by PrimozGodec** [comment]
It is now ready.

**Comment by ajdapretnar** [comment]
Hm. This doesn't seem to work on heart_disease... 🤔 

```
Traceback (most recent call last):
  File "/Users/ajda/miniconda3/envs/o3/lib/python3.7/site-packages/orangecanvas/scheme/signalmanager.py", line 718, in __process_next
    self.process_queued()
  File "/Users/ajda/miniconda3/envs/o3/lib/python3.7/site-packages/orangecanvas/scheme/signalmanager.py", line 476, in process_queued
    self.process_node(node_update_front[0])
  File "/Users/ajda/miniconda3/envs/o3/lib/python3.7/site-packages/orangecanvas/scheme/signalmanager.py", line 523, in process_node
    self.send_to_node(node, signals_in)
  File "/Users/ajda/miniconda3/envs/o3/lib/python3.7/site-packages/orangewidget/workflow/widgetsscheme.py", line 775, in send_to_node
    self.process_signals_for_widget(node, widget, signals)
  File "/Users/ajda/miniconda3/envs/o3/lib/python3.7/site-packages/orangewidget/workflow/widgetsscheme.py", line 809, in process_signals_for_widget
    handler(*args)
  File "/Users/ajda/orange/orange3/Orange/widgets/visualize/owboxplot.py", line 318, in set_data
    self.grouping_changed()
  File "/Users/ajda/orange/orange3/Orange/widgets/visualize/owboxplot.py", line 392, in grouping_changed
    self.attr_changed()
  File "/Users/ajda/orange/orange3/Orange/widgets/visualize/owboxplot.py", line 402, in attr_changed
    self.compute_box_data()
  File "/Users/ajda/orange/orange3/Orange/widgets/visualize/owboxplot.py", line 427, in compute_box_data
    dataset, attr, self.group_var)
  File "/Users/ajda/orange/orange3/Orange/statistics/contingency.py", line 286, in get_contingency
    dat, col_variable, row_variable, col_unknowns, row_unknowns)
  File "/Users/ajda/orange/orange3/Orange/statistics/contingency.py", line 194, in __init__
    self.from_data(dat, col_variable, row_variable)
  File "/Users/ajda/orange/orange3/Orange/statistics/contingency.py", line 220, in from_data
    conts = data._compute_contingency([col_variable], row_variable)
  File "/Users/ajda/orange/orange3/Orange/data/table.py", line 1490, in _compute_contingency
    col_data, classes_, n_rows, W_)
  File "Orange/data/_contingency.pyx", line 11, in Orange.data._contingency.contingency_floatarray
    @cython.wraparound(False)
ValueError: Buffer dtype mismatch, expected 'intp_t' but got 'double'
```

**Comment by PrimozGodec** [comment]
It works for me. There is a change in the Cython code. It was compiled in .c code already but it needs to be compiled again for each computer (at MacOS the result of the compilation is .so file). The easiest way to do that is by running:

    pip install -e . 

in the Orange folder when this PR is checked out.

**Comment by ajdapretnar** [comment]
Ah, important detail. 👌 Love the change!

**Pull request merged**